plugins/module_utils/resource.py

```diff
@@ -28,7 +28,10 @@
 def get_value(data, path):
     while path:
         key = path.pop(0)
-        data = data[key]
+        if isinstance(data, dict) and key in data:
+            data = data[key]
+        else:
+            raise KeyError(f"{data} is not of type dict or does not contain key {key}")
```

Comment:
Why hide a TypeError behind a KeyError? The KeyError is expected in the case of check mode, but if we are getting a TypeError here, I think that would indicate a bigger problem.